modules/worker_groups/main.tf

```diff
@@ -0,0 +1,241 @@
+resource "aws_autoscaling_group" "workers" {
+  for_each = local.worker_group_configurations
+
+  name_prefix = join(
+    "-",
+    compact(
+      [
+        var.cluster_name,
+        each.key,
+        each.value["asg_recreate_on_change"] ? random_pet.workers[each.key].id : ""
+      ]
+    )
+  )
+
+  desired_capacity        = each.value["asg_desired_capacity"]
+  max_size                = each.value["asg_max_size"]
+  min_size                = each.value["asg_min_size"]
+  force_delete            = each.value["asg_force_delete"]
+  target_group_arns       = each.value["target_group_arns"]
+  service_linked_role_arn = each.value["service_linked_role_arn"]
+  vpc_zone_identifier     = each.value["subnets"]
+  protect_from_scale_in   = each.value["protect_from_scale_in"]
+  suspended_processes     = each.value["suspended_processes"]
+
+  enabled_metrics = each.value["enabled_metrics"]
+
+  placement_group = each.value["placement_group"]
+
+  termination_policies      = each.value["termination_policies"]
+  max_instance_lifetime     = each.value["max_instance_lifetime"]
+  default_cooldown          = each.value["default_cooldown"]
+  health_check_grace_period = each.value["health_check_grace_period"]
+
+  dynamic mixed_instances_policy {
+    iterator = item
+    for_each = ((lookup(var.worker_groups[each.key], "override_instance_types", null) != null) || (lookup(var.worker_groups[each.key], "on_demand_allocation_strategy", null) != null)) ? list(each.value) : []
+
+    content {
+      instances_distribution {
+        on_demand_allocation_strategy            = lookup(item.value, "on_demand_allocation_strategy", "prioritized")
+        on_demand_base_capacity                  = item.value["on_demand_base_capacity"]
+        on_demand_percentage_above_base_capacity = item.value["on_demand_percentage_above_base_capacity"]
+
+        spot_allocation_strategy = item.value["spot_allocation_strategy"]
+        spot_instance_pools      = item.value["spot_instance_pools"]
+        spot_max_price           = item.value["spot_max_price"]
+      }
+
+      launch_template {
+        launch_template_specification {
+          launch_template_id = aws_launch_template.workers[each.key].id
+          version            = item.value["launch_template_version"]
+        }
+
+        dynamic "override" {
+          for_each = item.value["override_instance_types"]
+
+          content {
+            instance_type = override.value
+          }
+        }
+      }
+    }
+  }
+
+  dynamic launch_template {
+    iterator = item
+    for_each = ((lookup(var.worker_groups[each.key], "override_instance_types", null) != null) || (lookup(var.worker_groups[each.key], "on_demand_allocation_strategy", null) != null)) ? [] : list(each.value)
```

Comment:
does `var.worker_groups[each.key]` different from `each.value` ?

Reply:
This is the same as the previous case above